subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CopyTaskIntegrationSpec.groovy

```diff
@@ -2239,4 +2242,121 @@ class CopyTaskIntegrationSpec extends AbstractIntegrationSpec {
         "rename(Transformer)"       | "rename(org.gradle.internal.Transformers.noOpTransformer())"
     }
 
+    // region duplicates in compressed files
+    def "encountering duplicates in a zipTree vs an unzipped dir with DuplicateStrategy.FAIL should give a clear error"() {
+        given: "a directory with a file"
+        TestFile unzippedDir = file("before/files")
+        unzippedDir.file("sub/c.txt").createFile()
```

Comment:
💭 Style wise, I think `def`s look better here and `touch()` looks more familiar than `createFile()`